controllers/assignments.py

```diff
@@ -145,6 +145,7 @@ def get_summary():
         .select()
         .first()
     )
+
     res = db.executesql(
         """
     select chapter, name, min(score), max(score), to_char(avg(score), '00.999') as mean, count(score) from assignment_questions join questions on question_id = questions.id join question_grades on name = div_id
```

Comment:
If you add `question_type` to the list of columns then each row will already have the question type and you will not need to do another query on line 165.

Reply:
Are you talking about through the docstring starting at line 150 or somewhere else

Reply:
Line 150 is not a docstring.  It is a python string with an SQL query. -- a parameter to the `db.executesql` function on line 149.  And yes, this is what I was talking about.

Reply:
Looking more closely at the query I realized it is not as simple as just adding question_type to the select clause.  If a column is not in the group by clause you have to aggregate over it. In this case there is a one-to-one mapping from name to question type so we can use either min or max to aggregate and it will give us the same value.

this should do the trick...

```
select chapter, name, min(question_type) question_type, min(score), max(score), to_char(avg(score), '00.999') as  mean, count(score) 
from assignment_questions join questions on question_id = questions.id join question_grades on name =  div_id 
where assignment_id = %s and course_name = %s
group by chapter, name
```

Note that I used `min(question_type) question_type` so that the resulting column name will be called `question_type` and not `min`

Reply:
Thank you, that works better